File: cf-1971c.cpp
#include <iostream>
using namespace std;

int main () {
    int t;
    cin >> t;

    while (t--){
        int a, b, c, d;
        cin >> a >> b >> c >> d;

        bool isIntersect = false;
        int i = min(a, b), j = max(a, b);
        for (; i < j; i++) {
            if (c == i || d == i)
                isIntersect = !isIntersect;
        }

        cout << (isIntersect ? "YES" : "NO") << '\n';
    }
}

File: cf-1971c.py
t = int(input())

while t:
    t -= 1
    a, b, c, d = [int(i) for i in input().split()]
    
    m1, m2 = min(a, b), max(a, b)
    
    isIntersect = False
    for i in range(m1, m2):
        if i == c or i == d:
            isIntersect = not isIntersect
    
    if isIntersect:
        print("YES")
    else:
        print("NO")